[WIP] feat: V2.2 docs

smol-ninja opened the pull request:
Closes #134 and #133.

andreivladbrg commented on docs/contracts/v2/guides/05-hooks.md:
```diff
@@ -13,16 +13,24 @@ Sablier.
+| Hook                      | Arguments                                        | Description                                         |
+| ------------------------- | ------------------------------------------------ | --------------------------------------------------- |
+| `onLockupStreamCanceled`  | `(streamId,sender,senderAmount,recipientAmount)` | Called when the stream is canceled by the sender.   |
+| `onLockupStreamRenounced` | `(streamId)`                                     | Called when the stream is renounced by the sender.  |
```
`onLockupStreamRenounced` should be removed

andreivladbrg commented on docs/contracts/v2/guides/05-hooks.md:
```diff
@@ -13,16 +13,24 @@ Sablier.
+| `onLockupStreamRenounced` | `(streamId)`                                     | Called when the stream is renounced by the sender.  |
+| `onLockupStreamWithdrawn` | `(streamId,caller,to,amount)`                    | Called when an amount is withdrawn from the stream. |
+
+### Sender
```
sender hooks must be removed

andreivladbrg reviewed (commented):
i will leave some comments for now so that we can be more efficient

smol-ninja commented on docs/concepts/05-governance.md:
We should add a section on allow list but not sure if we should have governance. Its only one function. What does @PaulRBerg have to say about it?

smol-ninja replied:
Check this out: https://v2-docs-jwxolf6qa-sablier.vercel.app/concepts/protocol/hooks

PaulRBerg replied:
Oh we should definitely bring it back, as I have pointed out [here](https://github.com/sablier-labs/docs/pull/138/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#r1666026942).

smol-ninja replied:
Agree to bring the governance page back

smol-ninja commented on docs/concepts/protocol/04-tranches.md:
```diff
@@ -0,0 +1,39 @@
+| Amount   | `uint128` | The amount of assets to be unlocked in a tranche, denoted in units of the asset's decimals. |
+| Duration | `uint40`  | The time difference in seconds between this tranche and the previous one.                   |
+
+Each tranche has its own streaming function:
```
I didn't understand. Whats wrong with saying that it has a distribution function defined as: $$f(x) = \Sigma(eta)$$ where $\Sigma(eta)$ is the sum of all vested tranches' amounts?

smol-ninja replied:
I understood it. Yes I agree each tranche does not have a streaming function.

smol-ninja replied:
I realized it was not accurate for lockup dynamic too. Fixed in https://github.com/sablier-labs/docs/pull/138/commits/a18ffea092c5afee43808b3f21c797e88577732a

smol-ninja commented:
Thanks for giving it a quick look @andreivladbrg. I have addressed most of your feedbacks. I am going to do a thorough read today so its ready for review by tomorrow morning.

smol-ninja commented on docs/concepts/05-glossary.md:
```diff
@@ -81,11 +81,15 @@ A Lockup stream with a payment rate per second that can vary over time.
-Milestones play a crucial role in the calculation of the custom streaming curve.
+## Timestamp
+
+A timestamp is the time component of a segment or a tranche. It indicated the end of a particular segment or tranche.
```
Changed it to "A timestamp is the time component of a stream. In lockup linear, it represents start time, cliff time and end time whereas in lockup dynamic and lockup tranched, it plays a crucial role in the calculation of streaming curves.". Wdyt?

razgraf reviewed (approved):
Great stuff!

Any chance we can fold #154 into the mix (maybe after this gets merged, or into this?) as that's part of our v2.2 updates too. @smol-ninja

smol-ninja commented:
I will merge that in this PR.

PaulRBerg commented on docs/concepts/05-governance.md:
Why did you remove this? It has to be brought back.

- The NFT descriptor functionality still exists
- This is where the hook allowlist should be mentioned (maybe the full documentation for it should be elsewhere, but the feature should still be mentioned here)

smol-ninja replied:
I don't remember why I removed it but I guess I thought it is not required anymore since there is no protocol fee switch. I have brought it back now. 

On the allowlist, I have mentioned it at two places (ICYMI):
1. https://v2-docs-git-feat-v22-updates-sablier.vercel.app/concepts/protocol/hooks
2. https://v2-docs-git-feat-v22-updates-sablier.vercel.app/contracts/v2/guides/hooks

Also added to the glossary: https://v2-docs-git-feat-v22-updates-sablier.vercel.app/concepts/glossary#allow-list

PaulRBerg commented on docs/contracts/v2/reference/05-access-control.md:
```diff
@@ -4,32 +4,34 @@ sidebar_position: 5
 
-With the exception of the [admin functions](/docs/concepts/05-governance.md), all functionality in Sablier is related to
-one or more streams.
+With the exception of the [allow list](/contracts/v2/reference/core/abstracts/abstract.SablierV2Lockup#allowtohook), all
```
As per my other comment, the Governance document should be reinstantiated and this link should be updated accordingly.

smol-ninja replied:
https://github.com/sablier-labs/docs/pull/138/commits/debe1e613bf523b1f55a330acd3ce247cf09d196

PaulRBerg commented on docs/concepts/02-sablier-protocol.md:
```diff
@@ -15,12 +15,12 @@ under BUSL-1.1. Both protocol versions are open-source and can be accessed on Sa
 :::info
 
-Sablier is the first asset streaming protocol ever built in crypto, tracing its roots back to 2019.
+Sablier is the first money streaming protocol ever built, tracing its roots back to 2019.
```
we prefer to say "token streaming" and "token distribution" now.

will stick with "token streaming" in this particular context.

smol-ninja replied:
https://github.com/sablier-labs/docs/pull/138/commits/bffb33ee451c19488d0d603bc30fb7d5619bb106

PaulRBerg commented on docs/concepts/protocol/02-stream-types.mdx:
```diff
@@ -242,30 +235,43 @@ The Timelock shape locks all tokens for a specified period. Users cannot access
+for that use case. As a result, a stream with tranches created using Lockup Tranched is more gas efficient than the same
+stream created using Lockup Dynamic.
+
+These streams are powered by a number of user-provided "tranches", which is covered in the tranches article.
```
Let's link to the tranches article here.

smol-ninja replied:
https://github.com/sablier-labs/docs/pull/138/commits/bffb33ee451c19488d0d603bc30fb7d5619bb106

smol-ninja commented:
@PaulRBerg I have included your suggestions. lmk if it looks good now.

PaulRBerg commented on docs/concepts/protocol/06-airstreams.mdx:
```diff
@@ -70,23 +71,37 @@ in to prove that they are eligible, and create a stream if they are. That's all
 
+## Types of Airstreams
+
+We offer two types of Airstream campaigns:
```
In general, we should prefer non-personal formulations such as `There are` over `We offer`

cc @smol-ninja @andreivladbrg

smol-ninja replied:
I agree. "There are two types of Airstream campaigns:"?? If you are making changes already, feel free to change it.

smol-ninja replied:
https://github.com/sablier-labs/docs/pull/138/commits/eed3c918e5e832f4ed13a3380f96f1d8a7bf9dda

PaulRBerg reviewed (approved):
Stellar work!

Good to be merged.

smol-ninja commented:
Wait, I just found an issue.

smol-ninja commented on docs/concepts/05-governance.md:
```diff
@@ -36,6 +24,11 @@ wallets.
+| Blast           | [0xb1bEF51ebCA01EB12001a639bDBbFF6eEcA12B9F](https://blastscan.io/address/0xb1bEF51ebCA01EB12001a639bDBbFF6eEcA12B9F)            |
+=======
+| Blast           | [0xD427d37B5F6d33f7D42C4125979361E011FFbfD9](https://blastscan.io/address/0xD427d37B5F6d33f7D42C4125979361E011FFbfD9)            |
+>>>>>>> e16d76b (refactor: polish wording)
```
@PaulRBerg have we changed blast admin?

PaulRBerg replied:
Done

PaulRBerg commented:
Can you recheck @smol-ninja?

smol-ninja commented:
Yeah I am reviewing the commit.

smol-ninja commented:
We can merge now.

<img width="896" alt="LGTM" src="https://github.com/sablier-labs/docs/assets/6676622/841cc5e2-70a5-4b2b-8b20-4daee05505fc">
